iceoryx_hoofs/test/integrationtests/test_resizeable_lockfree_queue_stresstest.cpp

```diff
@@ -45,11 +46,58 @@ struct Data
     uint64_t count{0};
 };
 
+class Barrier
+{
+  public:
+    Barrier(uint32_t requiredCount = 0)
+        : m_requiredCount(requiredCount)
+    {
+    }
+
+    void notify()
+    {
+        uint32_t count;
+        {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            count = m_count.fetch_add(1) + 1;
+        }
+        if (count >= m_requiredCount)
+        {
+            m_condVar.notify_all();
+        }
+    }
+
+    void wait()
+    {
+        std::unique_lock<std::mutex> lock(m_mutex);
+        auto cond = [&]() { return m_count >= m_requiredCount; };
+        m_condVar.wait(lock, cond);
+    }
+
+    void reset(uint32_t requiredCount)
+    {
+        m_requiredCount = requiredCount;
+        m_count = 0;
+    }
+
+  private:
+    std::atomic<uint32_t> m_count{0};
```

Comment:
Does not need to be an atomic since it is guarded by a mutex

Reply:
Wanted to avoid the mutex but cannot with condvar.